samples/drivers/led_max7219/testcase.yaml

```diff
@@ -0,0 +1,13 @@
+sample:
+  name: led_max7219
+  description: Simple driver for max7219 led display.
```

Comment:
I see some discrepancy here. This is a sample application, not a driver. It actually shows how to use GPIO and SPI drivers to play with a MAX7219 device. I am not convinced that such application should be placed in `samples/drivers/led_max7219`. The part that actually drives the display should be rather extracted to a driver and placed in `drivers/led/max7219.c`, and this application should present how to utilize this driver.

Reply:
Well here's my reasoning behind this particular location.
When I started working on this I took some devel example which did something with SPI and started tinkering with it. The problem was that I couldn't find a simple sample which would show the ins and outs of using SPI in Zephyr. Most existing cases that have something to do with it, used it in a non-obvious way (at least to me as a beginner in this field).
Additionally I only wanted to make a simple example. Supporting more advanced usage scenarios (like chaining the devices together) would require work&testing which is outside of my current scope.
Therefore I'd rather stick to the current version and implement a standalone driver later (as another PR) and then refactor this code according to your suggestions.